fence/agents/compute/fence_compute.py

```diff
@@ -229,24 +229,20 @@ def fix_domain(options):
 				# Compute nodes are named as FQDN, strip off the hostname
 				calculated = service.host.replace(shorthost+".", "")
 
-			domains[calculated] = shorthost
-
 			if calculated == last_domain:
 				# Avoid complaining for each compute node with the same name
 				# One hopes they don't appear interleaved as A.com B.com A.com B.com
 				logging.debug("Calculated the same domain from: %s" % service.host)
+				continue
 
-			elif "--domain" in options and options["--domain"] == calculated:
-				# Supplied domain name is valid 
-				return
```

Comment:
I was thinking that we don't want to loop through all the compute nodes because there might be a LOT of them.
So I figured bailing early if we find at least one that had the supplied domain was sufficient